### PR TITLE
Introduced a new internal util to preserve relative format

### PR DESCRIPTION
**Pull request by quantumalaviya**
# What does this PR do?
Introduced a new internal util to preserve relative format

Discussed in #1006 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
@LukeWood

### CODE REVIEW

**Review by LukeWood — CHANGES_REQUESTED** [review]
If easy, lets also add a ValueError to ensure that `bounding_box_format` is in the list of formats.  you can use the converter keys() for this.

It may be good to add 1-2 unit tests too.  Thanks!

**Comment by quantumalaviya** [comment]
Sure, is there a list of supported `bounding_box_formats` declared somewhere in the repository? We wouldn't want to update this function every time a new format is added.

**Comment by quantumalaviya** [comment]
@LukeWood please check out the new tests and the ValueError

**Review by LukeWood — APPROVED** [review]
Looks good, thanks!

File: keras_cv/bounding_box/utils.py [hunk 1]
```diff
@@ -19,6 +19,26 @@
 from keras_cv.bounding_box.formats import XYWH
 
 
+def _preserve_rel(target_bounding_box_format, bounding_box_format):
```
Comment (LukeWood) [inline keras_cv/bounding_box/utils.py:22]: can we remove the _ here and just export the function?

File: keras_cv/bounding_box/iou.py [hunk 2]
```diff
@@ -115,10 +115,7 @@ def compute_iou(
             "or len(boxes2.shape)=3."
         )
 
-    if bounding_box_format.startswith("rel"):
-        target = "rel_yxyx"
-    else:
-        target = "yxyx"
+    target = bounding_box.utils._preserve_rel("yxyx", bounding_box_format)
```
Comment (LukeWood) [inline keras_cv/bounding_box/iou.py:118]: when using the function lets actually specify the argument names:

```
 target = bounding_box.utils._preserve_rel(target_format="yxyx", bounding_box_format=bounding_box_format)
```

**Comment by quantumalaviya** [comment]
@LukeWood Made the changes, please take a look now

**Comment by LukeWood** [comment]
hey @quantumalaviya LGTM but tests are failing!

**Pull request merged**